README.md

```diff
@@ -55,4 +55,34 @@ To run all the tests:
 grunt mochaTest
 ```
 
+#### Request Id logging
+`fh-loggger` also exports express-compatible middleware to generate unique requestId and automatically include it in logging methods.
 
+```
+var fh_logger = require('fh-logger');
+
+// must be called to setup the middleware
+var logger = fh_logger.createLogger({
+  name: 'first',
+  requestIdHeader: 'X-SOME-HTTP-HEADER'
+});
+app.use(logger.requestIdMiddleware);
+app.get('/', function(req, res) {
+  console.log(req.requestId) // -> uuid
+  logger.info('some message') // -> {msg: 'some message', requestId: 'some-uuid'}
+})
+```
+
+By default it utilizes the 'X-FH-REQUEST-ID' header, this can be overridden by the configuration passed to `createLogger` as shown above
+
+##### ensureRequestId
+
+For logging inside callbacks that are supposed to display the `requestId` but for some reason do not, utilize the exported `ensureRequestId({Function})`:
+
+```
+logger.ensureRequestId(function asyncOperation(err, data){
+  logger.error(err); // -> {requestId: 'some-uuid'}
+});
```

Comment:
This name is kind of ugly, but I was thinking about not utilizing terminology specific to `continuation-local-storage`, open to suggestions.